src/stave.ts

```diff
@@ -14,9 +14,69 @@ import { Clef } from './clef';
 import { KeySignature } from './keysignature';
 import { TimeSignature } from './timesignature';
 import { Volta } from './stavevolta';
+import { Bounds, Tempo, FactoryRendererOptions, FactoryStaveOptions, FontInfo } from './types/common';
+export interface StaveLineConfig {
+  visible: boolean;
+}
+
+export interface StaveOptions {
+  //[name: string]: any;
+  spacing: number;
+  thickness: number;
+  x_shift: number;
+  y_shift: number;
+  position_end?: number;
+  invert?: boolean;
+  cps?: { x: number; y: number }[];
+  bottom_text_position: number;
+  line_config: StaveLineConfig[];
+  space_below_staff_ln: number;
+  glyph_spacing_px: number;
+  space_above_staff_ln: number;
+  vertical_bar_width: number;
+  fill_style: string;
+  left_bar: boolean;
+  right_bar: boolean;
+  spacing_between_lines_px: number;
+  top_text_position: number;
+  num_lines: number;
+  size?: string;
+  renderer?: FactoryRendererOptions;
+  stave?: FactoryStaveOptions;
+  font?: FontInfo;
+  autoStem?: boolean;
+  secondaryBeamBreaks?: never[];
+  direction?: number;
+  harsh?: boolean;
+  superscript?: string;
+  position?: number;
+  style?: string;
```

Comment:
Many of these aren't stave options, not sure how they ended up here.

Reply:
removed